tests/TestSignature.cpp

```diff
@@ -107,92 +111,95 @@ static const char* rsa_1_public = "-----BEGIN RSA PUBLIC KEY-----\n"
 
 static QByteArray data("Some trivial test with a longer .... ................................. longer text");
 
+QSharedPointer<Botan::Private_Key> loadPrivateKey(const QString& pem)
+{
+    try {
+        std::string label;
+        auto der = Botan::PEM_Code::decode(pem.toStdString(), label);
+        auto key = new Botan::RSA_PrivateKey(
+            Botan::AlgorithmIdentifier("RSA", Botan::AlgorithmIdentifier::USE_NULL_PARAM), der);
+        return QSharedPointer<Botan::Private_Key>(key);
+    } catch (std::exception& e) {
+        qWarning("Failed to load key: %s", e.what());
```

Comment:
This is a test suite. You should QFAIL outa here.

Reply:
Since this is a helper function I cannot do that, but added a QVERIFY statement below each call to this function.